From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 27 Dec 2020 21:30:32 +0100
Subject: [PATCH] {BUG in 5.11-rc1} sound/soc/fsl/imx-hdmi: initialize ret
 before returning

clang on x86-64 reports:

    sound/soc/fsl/imx-hdmi.c:165:6: error: variable 'ret' is used
    uninitialized whenever 'if' condition is true
    [-Werror,-Wsometimes-uninitialized]
            if ((hdmi_out && hdmi_in) || (!hdmi_out && !hdmi_in)) {
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    sound/soc/fsl/imx-hdmi.c:212:9: note: uninitialized use occurs here
            return ret;
                   ^~~
    sound/soc/fsl/imx-hdmi.c:165:2: note: remove the 'if' if its
    condition is always false
            if ((hdmi_out && hdmi_in) || (!hdmi_out && !hdmi_in)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    sound/soc/fsl/imx-hdmi.c:165:6: error: variable 'ret' is used
    uninitialized whenever '||' condition is true
    [-Werror,-Wsometimes-uninitialized]
            if ((hdmi_out && hdmi_in) || (!hdmi_out && !hdmi_in)) {
                ^~~~~~~~~~~~~~~~~~~~~
    sound/soc/fsl/imx-hdmi.c:212:9: note: uninitialized use occurs here
            return ret;
                   ^~~
    sound/soc/fsl/imx-hdmi.c:165:6: note: remove the '||' if its
    condition is always false
            if ((hdmi_out && hdmi_in) || (!hdmi_out && !hdmi_in)) {
                ^~~~~~~~~~~~~~~~~~~~~~~~
    sound/soc/fsl/imx-hdmi.c:110:9: note: initialize the variable 'ret'
    to silence this warning
            int ret;
                   ^
                    = 0
---
 sound/soc/fsl/imx-hdmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/fsl/imx-hdmi.c b/sound/soc/fsl/imx-hdmi.c
index 2c2a76a71940..ede4a9ad1054 100644
--- a/sound/soc/fsl/imx-hdmi.c
+++ b/sound/soc/fsl/imx-hdmi.c
@@ -164,6 +164,7 @@ static int imx_hdmi_probe(struct platform_device *pdev)
 
 	if ((hdmi_out && hdmi_in) || (!hdmi_out && !hdmi_in)) {
 		dev_err(&pdev->dev, "Invalid HDMI DAI link\n");
+		ret = -EINVAL;
 		goto fail;
 	}
 
-- 
